src/imu/imu.cpp

```diff
@@ -77,6 +93,14 @@ bool getGyroscope(float *gyr_x, float *gyr_y, float *gyr_z)
 bool getMagnetometer(float *mag_x, float *mag_y, float *mag_z)
 {
 #if SF_PLATFORM == SF_PLATFORM_PARTICLE
+
+    myICM.getAGMT();
+
+    if (myICM.status != ICM_20948_Stat_Ok)
+    {
+        FLOG_AddError(FLOG_ICM_FAIL, 0);
+        return false;
+    }
     ICM_20948_AGMT_t agmt = myICM.getAGMT();
```

Comment:
Lines 97 and 104 appear to have duplicate behavior.  Would it be simpler to move line 104 earlier?